cantools/database/can/c_source.py

```diff
@@ -1530,14 +1526,9 @@ def _generate_definitions(database_name, messages, floating_point_numbers, use_f
 
                 signal_definitions.append(signal_definition)
 
-            if add_initializers and signal.initial:
-                init_value = float(signal.initial) * signal.scale + signal.offset
-                init_value = int(init_value) if not signal.is_float else float(init_value)
-                signals_init_body += INIT_SIGNAL_BODY_TEMPLATE_FMT.format(type_name=signal.type_name,
-                                                                          signal_initial=init_value,
-                                                                          signal_name=signal.snake_name,
-                                                                          signal_data_length=int(signal.type_length / 8)
-                                                                          )
+            if signal.initial:
+                signals_init_body += INIT_SIGNAL_BODY_TEMPLATE_FMT.format(signal_initial=signal.initial,
```

Comment:
... which makes this wrong, because `signal.initial` is the value of the signal in the physical world.
```suggestion
                signals_init_body += INIT_SIGNAL_BODY_TEMPLATE_FMT.format(signal_initial=signal.initial_raw,
```

Reply:
Yes, you're absolutely right, I made mistake yesterday.

Bit it led me to the following question:
Maybe it is also useful to have api for set/get signals in messages?

So that use case of initializing and further data accessing will look like this? 
(using **RT_IMU06_Accel** message from **vehicle.dbc** as example)
```
struct vehicle_rt_imu06_accel_t msg;
vehicle_rt_imu06_accel_init(struct vehicle_rt_imu06_accel_t *msg_p);
...
float physical_value = 100.0f;
vehicle_rt_imu06_accel_validity_accel_longitudinal_set_raw(&msg,
                                    vehicle_rt_imu06_accel_validity_accel_longitudinal_encode(physical_value));
...
float physical_value;
uint8_t raw_value;
vehicle_rt_imu06_accel_validity_accel_longitudinal_get_raw(&msg, &raw_value);
physical value =  vehicle_rt_imu06_accel_validity_accel_longitudinal_decode(raw_value);
```
What do you think about it?

Reply:
given that this is C, and C is supposed to be somewhat painful, I think it is not necessary: you can already assign physical values to a signal using
```C
msg.{signal_name} = {db_name}_{message_name}_{signal_name}_encode(1.234);
```

I think though, that in the long term a C++ generator should be added with an API like this:
```C++
msg.{signal_name}.assign_raw(1234);
msg.{signal_name}.assign(1.234);
```

I would want this codegen to use a proper template engine like [Jinja](https://jinja.palletsprojects.com), though...

Reply:
You're right, I wanted to decrease number of cases of explicit assignment to structure fields in user code

Reply:
Maybe modify api for decode like this?
```
float physical_value;
vehicle_rt_imu06_accel_validity_accel_longitudinal_get_real(&msg, &physical_value);
```
In this case function **vehicle_rt_imu06_accel_validity_accel_longitudinal_decode** is called internally.
I really think that explicitly accessing data members of object in user code is a good approach

Reply:
I don't really understand what the advantage of this would be compared to
```C
msg.validity_accel_longitudinal = vehicle_rt_imu06_accel_validity_accel_longitudinal_decode(physical_value);
```
(except for saving a few characters of boiler-plate. IMO that is a problem which is more or less intrinsic to plain-C.)

Reply:
Okay, I've understood your point of view.
Let's forget about setters/getters